### PR TITLE
V3.8.4 reduce code

### PR DESCRIPTION
**Pull request by star-e**
Simplify pipeline settings, to reduce code size.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

This pull request focuses on reducing code complexity and improving maintainability in the rendering pipeline and related components.

- **`cocos/rendering/custom/render-graph.ts`**: Added `isDepthStencil` parameter to `RenderSwapchain` class and methods for better swapchain configuration.
- **`cocos/rendering/custom/web-pipeline.ts`**: Modified `RenderSwapchain` instantiation in `WebPipeline` to include a new boolean parameter for enhanced depth-stencil handling.
- **`editor/assets/default_renderpipeline/builtin-pipeline-settings.ts`**: Replaced `makePipelineSettings` function with direct `PipelineSettings` instantiation and updated DoF properties to be hidden in the editor UI.
- **`editor/assets/default_renderpipeline/builtin-pipeline-types.ts`**: Refactored to use class-based definitions for rendering pipeline settings, improving readability and maintainability.
- **`native/cocos/renderer/pipeline/custom/FrameGraphDispatcher.cpp`**: Added `CC_EXPECTS` checks to ensure correct usage of depth-stencil properties in the rendering pipeline.

### CODE REVIEW

File: editor/assets/default_renderpipeline/builtin-pipeline-types.ts [hunk 1]
```diff
@@ -32,313 +32,159 @@ import { Material, Texture2D, gfx } from 'cc';
...
+    readonly colorGrading: ColorGrading = new ColorGrading();
+    readonly fsr: FSR = new FSR();
+    readonly fxaa: FXAA = new FXAA();
 }
 
 export function fillRequiredPipelineSettings(value: PipelineSettings): void {
-    if (value.msaa === undefined) {
-        (value.msaa as MSAA) = makeMSAA();
+    if (!value.msaa) {
+        (value.msaa as MSAA) = new MSAA();
```
Comment (dumganhar) [inline editor/assets/default_renderpipeline/builtin-pipeline-types.ts:155]: Why need to convert it to `MSAA` explictly?

File: editor/assets/default_renderpipeline/builtin-pipeline-types.ts [hunk 2]
```diff
@@ -32,313 +32,159 @@ import { Material, Texture2D, gfx } from 'cc';
...
+    value.shadingScale ??= 0.5;
+    if (!value.depthOfField) {
+        (value.depthOfField as DepthOfField) = new DepthOfField();
     } else {
         fillRequiredDepthOfField(value.depthOfField);
     }
-    if (value.bloom === undefined) {
-        (value.bloom as Bloom) = makeBloom();
+    if (!value.bloom) {
+        (value.bloom as Bloom) = new Bloom();
```
Comment (dumganhar) [inline editor/assets/default_renderpipeline/builtin-pipeline-types.ts:167]: I think `as Bloom` is not needed now.

File: editor/assets/default_renderpipeline/builtin-pipeline-types.ts (hunk 1 above)
Comment (star-e) [reply editor/assets/default_renderpipeline/builtin-pipeline-types.ts:155]: `value.msaa` is `readonly`, we cannot assign without the cast.

File: editor/assets/default_renderpipeline/builtin-pipeline-types.ts (hunk 2 above)
Comment (star-e) [reply editor/assets/default_renderpipeline/builtin-pipeline-types.ts:167]: The reason is the same as above.

Comment (dumganhar) [reply editor/assets/default_renderpipeline/builtin-pipeline-types.ts:167]: ok

File: editor/assets/default_renderpipeline/builtin-pipeline-types.ts (hunk 1 above)
Comment (dumganhar) [reply editor/assets/default_renderpipeline/builtin-pipeline-types.ts:155]: Got it.

**Review by dumganhar — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**